extension/arquillian-drone-appium-extension/src/main/java/org/arquillian/drone/appium/extension/webdriver/AppiumDriverFactory.java

```diff
@@ -0,0 +1,142 @@
+import io.appium.java_client.remote.MobileCapabilityType;
+import io.appium.java_client.remote.MobilePlatform;
+import io.appium.java_client.windows.WindowsDriver;
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.arquillian.config.descriptor.api.ArquillianDescriptor;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.drone.spi.Configurator;
+import org.jboss.arquillian.drone.spi.Destructor;
+import org.jboss.arquillian.drone.spi.DronePoint;
+import org.jboss.arquillian.drone.spi.Instantiator;
+import org.jboss.arquillian.drone.webdriver.configuration.WebDriverConfiguration;
+import org.jboss.arquillian.drone.webdriver.factory.CapabilitiesOptionsMapper;
+import org.jboss.arquillian.drone.webdriver.factory.ChromeDriverFactory;
+import org.jboss.arquillian.drone.webdriver.spi.BrowserCapabilities;
+import org.jboss.arquillian.drone.webdriver.spi.BrowserCapabilitiesRegistry;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import java.net.URL;
+
+import static org.arquillian.drone.appium.extension.webdriver.AppiumCapabilities.READABLE_NAME;
+
+/**
+ * Factory which combines {@link org.jboss.arquillian.drone.spi.Configurator},
+ * {@link org.jboss.arquillian.drone.spi.Instantiator} and {@link org.jboss.arquillian.drone.spi.Destructor} for
+ * Appium Java Client.
+ *
+ * @see <a href="https://github.com/appium/java-client">https://github.com/appium/java-client</a>
+ * @see <a href="https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md">https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md</a>
+ *  for supported capabilities
+ *
+ * @author Vaclav Muzikar <vmuzikar@redhat.com>
+ */
+public class AppiumDriverFactory implements
+    Configurator<AppiumDriver, WebDriverConfiguration>,
+    Instantiator<AppiumDriver, WebDriverConfiguration>,
+    Destructor<AppiumDriver> {
+
+    @Inject
+    private Instance<BrowserCapabilitiesRegistry> registryInstance;
+
+    @Override
+    public void destroyInstance(AppiumDriver instance) {
+        instance.quit();
+    }
+
+    @Override
+    public int getPrecedence() {
+        return 0;
+    }
+
+    /**
+     * Creates {@link AndroidDriver}, {@link IOSDriver}, {@link WindowsDriver} or generic {@link AppiumDriver}
+     * based on {@value MobileCapabilityType#PLATFORM_NAME} in Arquillian descriptor.
+     *
+     * @param configuration
+     *     the configuration object for the extension
+     *
+     * @return The Appium WebDriver
+     */
+    @Override
+    public AppiumDriver createInstance(WebDriverConfiguration configuration) {
+        Capabilities capabilities = getCapabilities(configuration);
+
+        String platform = (String)capabilities.getCapability(MobileCapabilityType.PLATFORM_NAME);
+        if (StringUtils.isBlank(platform)) {
```

Comment:
you can use our util methods: https://github.com/arquillian/arquillian-extension-drone/blob/master/drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/utils/Validate.java#L49
Then if this is the only place where you use `StringUtils` then you can remove it from dependencies.